src/middleware/routeHandler.ts

```diff
@@ -108,16 +108,19 @@ export const attachWriteRouteHandlerWrapper: RouteWrapper<{
     return
   }
 
-  Promise.resolve(routeHandler(req, res, next)).catch(async (err: Error) => {
-    await unlock(siteName)
-    next(err)
-  })
-
-  try {
-    await unlock(siteName)
-  } catch (err) {
-    next(err)
-  }
+  Promise.resolve(routeHandler(req, res, next)).then(
+    async () => {
+      try {
+        await unlock(siteName)
+      } catch (err) {
+        next(err)
+      }
+    },
+    async (err) => {
+      await unlock(siteName)
+      next(err)
```

Comment:
Same vein as above: comparatively, why is this unlock not in a try..catch? This is the catch handler so the handler failed (and presumably did not call `next()`, so if the lock release also fails, should we not guarantee a call to `next(err)`?